File: backend/app/schemas/api_config.py
"""
用户 API 配置 Pydantic Schema
"""
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class ApiConfigOut(BaseModel):
    """API 配置输出（key 脱敏）"""
    provider: str = "deepseek"
    api_key_masked: str = ""
    base_url: str = ""
    model: str = ""
    is_active: bool = True
    is_custom: bool = False
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None


class ApiConfigSaveIn(BaseModel):
    """保存/更新 API 配置请求"""
    provider: str = Field(default="deepseek", description="提供商: deepseek | openai | custom")
    api_key: str = Field(..., min_length=1, max_length=256, description="API Key")
    base_url: str = Field(default="https://api.deepseek.com", max_length=512, description="API 端点")
    model: str = Field(default="deepseek-v4-flash", max_length=128, description="模型名称")


class ApiConfigTestIn(BaseModel):
    """测试连接请求（key 不存储）"""
    provider: str = Field(default="deepseek", pattern="^(deepseek|openai|custom)$")
    api_key: str = Field(..., min_length=1, max_length=512)
    base_url: str = Field(default="https://api.deepseek.com", max_length=512)
    model: str = Field(default="deepseek-v4-flash", max_length=128)
    timeout_seconds: int = Field(default=60, ge=5, le=600)
    max_retries: int = Field(default=2, ge=0, le=5)
    temperature: float = Field(default=0.2, ge=0, le=2)


class ApiConfigTestOut(BaseModel):
    """测试连接结果"""
    success: bool
    message: str
    model: str = ""
    duration_ms: int = 0
    attempts: int = 0
    retryable: bool = False
    next_action: str = ""

File: frontend/src/api/apiConfig.ts
import { get, post, put, del } from './http'

export interface ApiConfigOut {
  provider: string
  api_key_masked: string
  base_url: string
  model: string
  is_active: boolean
  is_custom: boolean
  created_at?: string
  updated_at?: string
}

export interface ApiConfigSaveIn {
  provider: string
  api_key: string
  base_url: string
  model: string
}

export interface ApiConfigTestIn {
  provider: string
  api_key: string
  base_url: string
  model: string
}

export interface ApiConfigTestOut {
  success: boolean
  message: string
  model: string
  duration_ms: number
}

export function getApiConfig(): Promise<ApiConfigOut> {
  return get<ApiConfigOut>('/api-config')
}

export function saveApiConfig(payload: ApiConfigSaveIn): Promise<ApiConfigOut> {
  return put<ApiConfigOut>('/api-config', payload)
}

export function deleteApiConfig(): Promise<string> {
  return del<string>('/api-config')
}

export function testApiConnection(payload: ApiConfigTestIn): Promise<ApiConfigTestOut> {
  return post<ApiConfigTestOut>('/api-config/test', payload)
}
